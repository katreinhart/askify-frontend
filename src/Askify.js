import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Redirect, withRouter } from 'react-router-dom'
import SignUp from './components/SignUp'
import SignIn from './components/SignIn'
import Queue from './components/Queue'
import QuestionArchive from './components/QuestionArchive'

import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
  signIn,
  signUp,
  fetchUser,
  updateQueuePosition
} from './actions/user.actions'

import { fetchQueue } from './actions/queue.actions'

import { fetchArchive } from './actions/archive.actions'


const isPrivate = ({ id }, Component) => {
  console.log(id);
  return (props) => id ? <Component {...props} /> : <Redirect to="/"/>
}

class Askify extends Component {
  constructor(props){
    super(props)
  }

  shouldComponentUpdate() {
    const position = this.props.queue.findIndex((item => {
      return item.userid === this.props.user.id
    }))
    if (position !== this.props.user.order) {
      return true
    } else {
      return false
    }
  }

  async componentDidMount () {
    const token = await window.localStorage.getItem('askifyToken')
    if(token) {
      const user = await this.props.fetchUser(token)
      const queue = await this.props.fetchQueue(token)
      const archive = await this.props.fetchArchive(token)
      this.props.updateQueuePosition(this.props.user, this.props.queue)
    }
  }

  render() {
    return (
      <div>
        <Router>
          <div className="App">
            <Route exact path='/' component={() => this.props.user.id ?
              <Redirect to="/queue"/> : <SignIn />} />
            <Route path='/signup' component={() => this.props.user.id ?
              <Redirect to="/queue"/> : <SignUp />} />
            <Route path='/queue' component={ isPrivate(this.props.user, Queue) } />
            <Route path='/archive' component={ isPrivate(this.props.user, QuestionArchive) } />
          </div>
        </Router>
      </div>
    );
  }
}

function mapStateToProps (state) {
  return {
    queue: state.queue,
    archive: state.archive,
    user: state.user,
  }
}

function mapDispatchToProps (dispatch) {
  return bindActionCreators({
    fetchUser,
    signIn,
    signUp,
    fetchQueue,
    updateQueuePosition,
    fetchArchive
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Askify)
