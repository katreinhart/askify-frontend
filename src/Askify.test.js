import React from 'react';
import { mount, shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Askify } from './Askify';

describe('Askify', () => {
  let askify = shallow(<Askify />)

  const createMockStore = configureMockStore([thunk])
  const store = createMockStore({ user: {} })

  it('renders without crashing', () => {
    expect(askify).toMatchSnapshot()
  })

  // describe('when there is no user logged in', () => {
  //   let askify = mount(<Askify />)
  //   it('renders sign in page', () => {
  //     expect(askify.find('Connect(SignIn)').exists()).toBe(true)
  //   })
  // })
})
