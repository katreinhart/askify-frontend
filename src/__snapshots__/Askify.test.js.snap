// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Askify renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Askify />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <div
                className="App"
        >
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        path="/signup"
                />
                <Route
                        component={[Function]}
                        path="/queue"
                />
                <Route
                        component={[Function]}
                        path="/archive"
                />
        </div>
</BrowserRouter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/signup"
          />
          <Route
                    component={[Function]}
                    path="/queue"
          />
          <Route
                    component={[Function]}
                    path="/archive"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/signup"
/>,
            <Route
              component={[Function]}
              path="/queue"
/>,
            <Route
              component={[Function]}
              path="/archive"
/>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/signup",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/queue",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/archive",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <div
                    className="App"
          >
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              path="/signup"
                    />
                    <Route
                              component={[Function]}
                              path="/queue"
                    />
                    <Route
                              component={[Function]}
                              path="/archive"
                    />
          </div>
</BrowserRouter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="App"
>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/signup"
            />
            <Route
                        component={[Function]}
                        path="/queue"
            />
            <Route
                        component={[Function]}
                        path="/archive"
            />
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/signup"
/>,
              <Route
                component={[Function]}
                path="/queue"
/>,
              <Route
                component={[Function]}
                path="/archive"
/>,
            ],
            "className": "App",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/queue",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/archive",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
